jmclient/jmclient/blockchaininterface.py

```diff
@@ -167,22 +180,16 @@ def __init__(self, jsonRpc, network):
         actualNet = blockchainInfo['chain']
 
         netmap = {'main': 'mainnet', 'test': 'testnet', 'regtest': 'regtest'}
-        if netmap[actualNet] != network:
+        if netmap[actualNet] != network and \
+                (not (actualNet == "regtest" and network == "testnet")):
+            #special case of regtest and testnet having the same addr format
```

Comment:
I don't understand, has anything changed here? Surely the output of getblockchaininfo hasn't, so if we still enter the same values for 'network' in the config file, why would this need an edit?

Reply:
This is needed to test `bitcoin-rpc-no-history` on regtest.

The parameter `network` doesn't always come from the `network` value in the config file. Previously if you use regtest the only possible `blockchain_source` was `regtest`. In that case the constructor of `RegtestBitcoinCoreInterface` would set the `network` parameter in this function to be `regtest`.

But after this PR `bitcoin-rpc-no-history` is a way to use regtest too. That's what changed.

Now, the only acceptable values for `network` in the config file are `mainnet` and `testnet`, they are used to determine what vbytes to use in bitcoin addresses, so I decided against adding `regtest` as an option there because its exactly the same vbytes as testnet. Also it would be far more lines to edit, much of which would just be duplicates. Although there is an argument that it would be more correct to add regtest to cryptoengine.py

Reply:
Yes I see. Damn it's complicated (and the code is kinda weird, haven't looked at it for years I guess). RegtestBitcoinCoreInterface passes network value 'testnet' into BitcoinCoreInterface constructor; others (before this PR) set the network value from the config. While if BitcoinCoreNoHistoryInterface wants to allow both networks (regtest and testnet) you have to allow the additional possibility (with BitcoinCoreInterface doesn't because it explicitly has the different Regtest... class).
Oh well this can be cleaned up at some point to be less confusing.